src/runner.ts

```diff
@@ -51,28 +51,33 @@ export class Runner {
       const suite = new RunnerSuite(rootFixtures, '');
       suite.file = file;
       const revertBabelRequire = runnerSpec(suite, config);
+      let cleanup = () => void 0;
       try {
-        require(file);
+        cleanup = requireWithCleanup(file);
       } catch (e) {
         prependErrorMessage(e, `Error while reading ${file}:\n`);
         throw e;
       }
+      if (localOverrideParameterRegistrations.size)
+        cleanup();
```

Comment:
This is kind of expensive, because we have to re-require all the test dependencies (playwright) every time a parameter changes. I have a bunch of fancy heuristic magic that makes this pretty much free while always being correct. If performance concerns are blocking this, I'll send it in a standalone patch.